### PR TITLE
Added cli

### PR DESCRIPTION
**Pull request by Cr0a3**
Hi,
I added cli options.
The help message:
image
The version message:
image

I also fixed a possible segfault when passing in unknown files.
I also  added a dump method wich dumps all of the register values and names.
Bye

### CODE REVIEW

File: src/asm/runner.c [hunk 1]
```diff
@@ -27,15 +27,16 @@ static void run_ins(VirtualMachine *vm, CasmElement elem) {
...
-      Value val = op_to_val(&elem.var.ins.args[0]);
-      move(vm, val, op_to_dest(&elem.var.ins.args[1]));
-      break;
-    }
-    case AST_INS_SYSCALL: {
-      syscall(vm);
-      break;
-    }
+      case AST_INS_MOV: {
+        printf("mov");
```
Comment (Thepigcat76) [inline src/asm/runner.c]: Is this print for debugging? If so, please remove it

File: src/cli.h [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#ifndef CLI_H
```
Comment (Thepigcat76) [inline src/cli.h]: please change header guards to use `#pragma once` instead of defines

File: src/vm/vm.c [hunk 3]
```diff
@@ -81,4 +81,14 @@ void syscall(VirtualMachine *vm) {
     syscall_print(vm);
     break;
   }
+}
+
+#include "asm/ast.h"
```
Comment (Thepigcat76) [inline src/vm/vm.c]: maybe move the reg_to_string(...) function to vm.c so the asm parsing stuff and actual virtual machine stay seperate

**Comment by Thepigcat76** [comment]
Thanks for your pull request! Didn't expect any interaction on this project since it was just for fun but I appreciate it nonetheless. I reviewed your changes and overall they are really good! Just some minor things that I feel should be changed.

File: src/asm/ast.h [hunk 4]
```diff
@@ -12,6 +12,8 @@ typedef enum {
   RA3,
 } Register;
 
+char* reg_to_string(int num);
```
Comment (Thepigcat76) [inline src/asm/ast.h]: As i said, consider moving this to vm.c and possibly even making it static

File: src/vm/vm.h [hunk 5]
```diff
@@ -70,3 +70,5 @@ void add(VirtualMachine *vm, uint64_t val0, uint64_t val1, Dest dest);
 void move(VirtualMachine *vm, Value val, Dest dest);
 
 void syscall(VirtualMachine *vm);
+
+void dump(VirtualMachine vm);
```
Comment (Thepigcat76) [inline src/vm/vm.h]: Consider passing the VirtualMachine as a pointer

**Review by Thepigcat76 — CHANGES_REQUESTED** [review]

**Comment by Cr0a3** [comment]
@Thepigcat76 fixed it

File: src/vm/vm.h [hunk 6]
```diff
@@ -71,4 +71,7 @@ void move(VirtualMachine *vm, Value val, Dest dest);
 
 void syscall(VirtualMachine *vm);
 
-void dump(VirtualMachine vm);
\ No newline at end of file
+void dump(VirtualMachine *vm);
+
+
+static char* reg_to_string(int num);
```
Comment (Thepigcat76) [inline src/vm/vm.h:77]: Static functions are essentially private, so there is no use in exposing them through a header file. Could you please remove this?

Comment (Thepigcat76) [inline src/vm/vm.h]: Static functions are essentially private, so there is no use in exposing them through a header file. Could you please remove this?

**Comment by Thepigcat76** [comment]
Thank you very much! Only one minor nitpick and then this can be merged :+1:

**Comment by Cr0a3** [comment]
> Only one minor nitpick and then this can be merged 👍

Which one?

**Comment by Cr0a3** [comment]
> Only one minor nitpick

Which one?

**Comment by Thepigcat76** [comment]
image
This one, above my message

**Comment by Cr0a3** [comment]
Sorry fixed it

**Review by Thepigcat76 — APPROVED** [review]

**Comment by Thepigcat76** [comment]
No need to apologize, you did an amazing job with this pr! And btw, you have my biggest respect for being this young and still having this much knowledge :+1:

**Comment by Cr0a3** [comment]
Thank you very much

**Pull request merged**